contextproject-ms3/src/main/java/contextproject/models/Track.java

```diff
@@ -42,8 +47,60 @@ public Track(String abPath) {
       log.error("There was a IO exception with file:" + abPath);
       log.trace(StackTrace.stackTrace(e));
     }
-    absolutePath = abPath;
+    this.absolutePath = abPath;
+    this.title = info.get("title");
+    this.artist = info.get("artist");
+    this.album = info.get("album");
+
+    try {
+      this.length = Long.parseLong(info.get("length"));
+    } catch (NumberFormatException e) {
+      this.length = song.getLengthInMilliseconds();
+    }
+
+    try {
+      this.bpm = Double.parseDouble((info.get("bpm")));
+    } catch (NullPointerException | NumberFormatException f) {
+      this.bpm = 0.0;
+    }
+
+    if (info.get("key") != null) {
+      this.key = new Key(info.get("key"));
+    }
+
     getMetadata();
+
+    int startBeatIntro;
+    int introBeatLength;
+    int startBeatOurto;
+    int outroBeatLength;
+    long firstBeat;
+
+    try {
+      startBeatIntro = Integer.parseInt(info.get("startBeatIntro"));
+      introBeatLength = Integer.parseInt(info.get("introBeatLength"));
+    } catch (NumberFormatException e) {
+      startBeatIntro = 0;
+      introBeatLength = 0;
+    }
+
+    try {
+      startBeatOurto = Integer.parseInt(info.get("startBeatOutro"));
+      outroBeatLength = Integer.parseInt(info.get("outroBeatLength"));
+    } catch (NumberFormatException e) {
+      startBeatOurto = 0;
+      outroBeatLength = 0;
+    }
+
+    try {
+      firstBeat = Long.parseLong(info.get("firstBeat"));
+    } catch (NumberFormatException e) {
+      firstBeat = 0;
+    }
+
+    this.beatGrid = new BeatGrid(this.length, this.bpm, firstBeat, startBeatIntro, introBeatLength,
+        startBeatOurto, outroBeatLength);
+
   }
 
```

Comment:
Isn't it better to call a method in the constructor that does all the work? It looks pretty messy to have a constructor with that many lines of code.